modules/afmongodb/afmongodb-worker.c

```diff
@@ -388,17 +482,38 @@ _worker_insert(LogThreadedDestWorker *s, LogMessage *msg)
             evt_tag_value_pairs("message", owner->vp, msg, &options),
             evt_tag_str("driver", owner->super.super.super.id));
 
-
   if (!owner->collection_is_literal_string)
     {
+      g_string_assign(self->last_collection, self->collection->str);
```

Comment:
last_collection is not really useful in the MongoDB worker struct, as it is local information needed for a very short time.

`scratch_buffers_alloc_and_mark()` or `scratch_buffers_alloc()` can be used to achieve the same performance.